src/librustc/session/config.rs

```diff
@@ -223,6 +223,15 @@ impl OutputFilenames {
     pub fn filestem(&self) -> String {
         format!("{}{}", self.out_filestem, self.extra)
     }
+
+    pub fn is_path_used(&self, path: PathBuf) -> bool {
+        for flavor in self.outputs.keys() {
+            if self.path(*flavor) == path {
```

Comment:
There is a `.iter()` method that directly returns an iterator over (key, value) pairs. In fact, in this context, I think you do not even care about keys and could even use `.values()` to iterate over values only instead.